src/main/java/org/springframework/data/convert/CustomConversions.java

```diff
@@ -0,0 +1,489 @@
+	private final Map<ConvertiblePair, Optional<Class<?>>> customWriteTargetTypes;
+	private final Map<Class<?>, Optional<Class<?>>> rawWriteTargetTypes;
+
+	/**
+	 * Creates a new {@link CustomConversions} instance registering the given converters.
+	 * @param converters
+	 */
+	public CustomConversions(StoreConversions storeConversions, List<?> converters) {
+
+		Assert.notNull(converters, "List of converters must not be null!");
+
+		this.readingPairs = new LinkedHashSet<>();
+		this.writingPairs = new LinkedHashSet<>();
+		this.customSimpleTypes = new HashSet<>();
+		this.customReadTargetTypes = new ConcurrentHashMap<>();
+		this.customWriteTargetTypes = new ConcurrentHashMap<>();
+		this.rawWriteTargetTypes = new ConcurrentHashMap<>();
+
+		List<Object> toRegister = new ArrayList<Object>();
+
+		// Add user provided converters to make sure they can override the defaults
+		toRegister.addAll(converters);
+		toRegister.addAll(JodaTimeConverters.getConvertersToRegister());
+		toRegister.addAll(Jsr310Converters.getConvertersToRegister());
+		toRegister.addAll(ThreeTenBackPortConverters.getConvertersToRegister());
+		toRegister.addAll(storeConversions.getStoreConverters());
+
+		toRegister.stream()//
+				.flatMap(it -> storeConversions.getRegistrationsFor(it).stream())//
+				.forEach(this::register);
+
+		Collections.reverse(toRegister);
+
+		this.converters = Collections.unmodifiableList(toRegister);
+		this.simpleTypeHolder = new SimpleTypeHolder(customSimpleTypes, storeConversions.getStoreTypeHolder());
+	}
+
+	/**
+	 * Returns the underlying {@link SimpleTypeHolder}.
+	 * 
+	 * @return
+	 */
+	public SimpleTypeHolder getSimpleTypeHolder() {
+		return simpleTypeHolder;
+	}
+
+	/**
+	 * Returns whether the given type is considered to be simple. That means it's either a general simple type or we have
+	 * a writing {@link Converter} registered for a particular type.
+	 * 
+	 * @see SimpleTypeHolder#isSimpleType(Class)
+	 * @param type
+	 * @return
+	 */
+	public boolean isSimpleType(Class<?> type) {
+
+		Assert.notNull(type, "Type must not be null!");
+
+		return simpleTypeHolder.isSimpleType(type);
+	}
+
+	/**
+	 * Populates the given {@link GenericConversionService} with the converters registered.
+	 * 
+	 * @param conversionService
+	 */
+	public void registerConvertersIn(GenericConversionService conversionService) {
+
+		Assert.notNull(conversionService, "ConversionService must not be null!");
+
+		converters.forEach(it -> {
+
+			boolean added = false;
+
+			if (it instanceof Converter) {
+				conversionService.addConverter(Converter.class.cast(it));
+				added = true;
+			}
+
+			if (it instanceof ConverterFactory) {
+				conversionService.addConverterFactory(ConverterFactory.class.cast(it));
+				added = true;
+			}
+
+			if (it instanceof GenericConverter) {
+				conversionService.addConverter(GenericConverter.class.cast(it));
+				added = true;
+			}
+
+			if (!added) {
+				throw new IllegalArgumentException(
```

Comment:
Exposing the offending instance is a good idea. Is there added value in logging the converters registered? If so, shouldn't this be functionality `GenericConversionService`?